mne/preprocessing/ica.py

```diff
@@ -962,6 +954,101 @@ def get_components(self):
+            jointly across all supplied components. If ``None`` (default), uses
+            all available components.
+
+        Returns
+        -------
+        float
+            The fraction of variance in ``inst`` that can be explained by the
+            ICA components.
+
+        Notes
+        -----
+        Since ICA components cannot be assumed to be aligned orthogonally, the
+        sum of the proportion of variance explained by all components may not
+        be equal to 1. In certain edge cases, the proportion of variance
+        explained by a component may even be negative.
+
+        .. versionadded:: 1.2
+        """
+        _validate_type(
+            item=inst, types=(BaseRaw, BaseEpochs, Evoked),
+            item_name='inst'
+        )
+        _validate_type(
+            item=components, types=(None, 'int-like', Sequence, np.ndarray),
+            item_name='components'
+        )
+        if isinstance(components, (Sequence, np.ndarray)):
+            for item in components:
+                _validate_type(
+                    item=item, types='int-like',
+                    item_name='Elements of "components"'
+                )
+
+        if self.current_fit == 'unfitted':
+            raise ValueError('ICA must be fitted first.')
+
+        if components is None:
+            components = range(self.n_components_)
+
+        # The algorithm implemented below should be equivalent to
+        # https://sccn.ucsd.edu/pipermail/eeglablist/2014/009134.html
+        #
+        # Reconstruct ("back-project") the data using only the specified ICA
+        # components. Don't make use of potential "spare" PCA components in
+        # this process – we're only interested in the contribution of the ICA
+        # components!
+        kwargs = dict(
+            inst=inst.copy(),
+            include=[components],
+            exclude=[],
+            n_pca_components=0,
+            verbose=False,
+        )
+        if (
+            isinstance(inst, (BaseEpochs, Evoked)) and
+            inst.baseline is not None
+        ):
+            # Don't warn if data was baseline-corrected.
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
+                    action='ignore',
+                    message='The data.*was baseline-corrected',
+                    category=RuntimeWarning
+                )
+                inst_recon = self.apply(**kwargs)
+        else:
+            inst_recon = self.apply(**kwargs)
+
+        data_recon = inst_recon.get_data(picks=self.ch_names)
+        data_orig = inst.get_data(picks=self.ch_names)
+        data_diff = data_orig - data_recon
+
+        # To estimate the data variance, we first compute the variance across
+        # channels at each time point, and then we average these variances.
+        mean_var_diff = data_diff.var(axis=0).mean()
+        mean_var_orig = data_orig.var(axis=0).mean()
```

Comment:
I worry a bit about this approach because it is going to be very sensitive to channel scaling. If you use a prewhitener in MNE MEG+EEG can be processed jointly (I think?), but this `var` calculation will make EEG the only thing that matters.

I think you can fix this by applying the pre-whitener to the `data_diff` and the `data_orig` before this calculation, then everything might "just work". But @agramfort would know better than me...

Reply:
Excellent point. 

How about we make our life's easy and simply apply this algorithm here to each channel,type,separately, and return a diet with the results? One like we use for `reject` and `flat` when creating epochs?

Reply:
Yes I think we have to do this. As it makes no sense compute variance with mixed channel types.

Reply:
This is now implemented.